bin/backup.sh

```diff
@@ -18,17 +18,13 @@ if [[ -z "$S3_BUCKET_PATH" ]]; then
   exit 1
 fi
 
-#install aws-cli
-curl https://s3.amazonaws.com/aws-cli/awscli-bundle.zip -o awscli-bundle.zip
-unzip awscli-bundle.zip
-chmod +x ./awscli-bundle/install
-./awscli-bundle/install -i /tmp/aws
+DATABASE_NAME="$(/app/vendor/heroku-toolbelt/bin/heroku pg:info -a $APP | grep DATABASE_URL | awk '{split($0,a," "); print a[2]}')"
```

Comment:
@kbaum, Actually I am doing the same thing as you people are doing. I am doing backup of my production heroku app database from development heroku app. 

Initially I tried with creating a new heroku app, but it seems heroku schedular needs credit card info before being added. So I decided to use our existing heroku development app to do the backup. 

But as the app already being created, I have to follow some extra steps to make it work. 

I think I made this line coupled with DATABASE_URL, we can go with $DATABASE like below,
then you can provide DATABASE=HEROKU_VIOLET_*** or DATABASE_URL both. 

+DATABASE_NAME="$(/app/vendor/heroku-toolbelt/bin/heroku pg:info -a $APP | grep $DATABASE | awk '{split($0,a," "); print a[2]}')"

About your separate application concern, $APP defines which app we need to backup, So as you specify the remote app in $APP and remote db in $DATABASE you are good go. 

In my schedular command, I invoke the script as DATABASE=DATABASE_URL APP=dev_app ... because DATABASE_URL always points to current DATABASE of the APP.  So, I don't have to change the DB name in the script each time when DATABASE has changed.